esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -1935,30 +1935,41 @@ void EspHttpBinding::generateSampleXmlFromSchema(bool isRequest, IEspContext &co
     StringBuffer element, schemaXmlbuff(schemaxml);
     getXMLMessageTag(context, isRequest, methodQName.str(), element);
 
-    Owned<IXmlSchema> schema = createXmlSchema(schemaXmlbuff);
-    if (schema.get())
+    StringBuffer content("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+    if (context.queryRequestParameters()->hasProp("display"))
+        content.append("<?xml-stylesheet type=\"text/xsl\" href=\"/esp/xslt/xmlformatter.xsl\"?>");
+
+    if (!methodQName || !*methodQName)
     {
-        IXmlType* type = schema->queryElementType(element);
-        if (type)
+        StringBuffer label(isRequest ? "Request" : "Response");
```

Comment:
In the native services these tags are plural "Requests", "Responses".

Reply:
The JSON generation returns "Request" and "Response" (line 1912). Should we change them as well?

Reply:
Ah, I see that.  In some ways it's inconsistent, but yes, in native services it is plural in xml, and singular in JSON.  So I would make it plural here.

Btw, the reason is that in xml it is a root element that contains muliple children of different names.  So <requests><a/><b/><c/>  vs. In JSON it is an array. request[1], request[2].

Reply:
Understood, @afishbeck. Thanks for the explanation. I have just pushed your suggested changes.